File: e-book-repository/src/main/java/com/ftn/service/UserService.java
package com.ftn.service;

import com.ftn.model.dto.PasswordDTO;
import com.ftn.model.dto.UserDTO;

import java.util.List;

/**
 * Created by Jasmina on 16/06/2018.
 */
public interface UserService {

    UserDTO readSelf();

    List<UserDTO> readAll();

    List<UserDTO> readSubscribers();

    List<UserDTO> readSubscribersTo(Long categoryId);

    UserDTO create(UserDTO userDTO);

    UserDTO update(Long id, UserDTO userDTO);

    boolean updatePassword(Long id, PasswordDTO passwordDTO);

    void subscribe(Long userId, Long categoryId);

    void cancelSubscription(Long id);

    void delete(Long id);

    boolean uniqueUsername(String username);
}

File: e-book-repository/src/main/webapp/services/userService.js
/**
 * Created by Jasmina on 13/07/2018.
 */
app.service('userService', function($http){
    return {
        read: function(onSuccess, onError){
            $http.get('/api/users').then(onSuccess, onError);
        },
        readMe: function (onSuccess, onError) {
            $http.get('/api/users/me').then(onSuccess, onError);
        },
        readSubs: function (onSuccess, onError) {
            $http.get('/api/users/subs').then(onSuccess, onError);
        },
        create: function(user, onSuccess, onError){
            $http.post('/api/users', user).then(onSuccess, onError);
        },
        update: function(user, onSuccess, onError){
            $http.patch('/api/users/' + user.id, user).then(onSuccess, onError);
        },
        delete: function (id, onSuccess, onError) {
            $http.delete('/api/users/' + id).then(onSuccess, onError);
        },
        subscribe: function (userId, categoryId, onSuccess, onError) {
            $http.post('/api/users/' + userId + '/' + categoryId).then(onSuccess, onError);
        }
    }
});